test/roby_app/test_plugin.rb

```diff
@@ -212,6 +212,143 @@ def perform_app_assertion(result)
                     perform_app_assertion interface.unit_tests.orogen_deployment_exists?
                 end
             end
+
+            describe "Local Log Transfer Server" do
+                def pre_config_server
+                    @app = Roby::Application.new
+                    @app.log_dir = make_tmpdir
+                    @current_log_level = RemoteProcesses::Server.logger.level
+                    RemoteProcesses::Server.logger.level = Logger::FATAL + 1
+                end
+
+                def start_server
+                    raise "server already started" if @server
+
+                    @server = RemoteProcesses::Server.new(@app, port: 0)
+                    server.open
+                    @server_thread = Thread.new { server.listen }
+                end
+
+                def connect_to_server
+                    @root_loader = OroGen::Loaders::Aggregate.new
+                    OroGen::Loaders::RTT.setup_loader(root_loader)
+                    @client = RemoteProcesses::Client.new(
+                        "localhost",
+                        server.port,
+                        root_loader: root_loader
+                    )
+                end
+
+                def start_and_connect_to_server
+                    start_server
+                    connect_to_server
+                end
+
+                def start_process_server(name, host)
+                    @ps_log_dir = make_tmpdir
+                    @client.create_log_dir(
+                        @ps_log_dir, Roby.app.time_tag,
+                        { "parent" => Roby.app.app_metadata }
+                    )
+                    Configuration.register_process_server(name, @client, log_dir, host_id: name)
+                end
+
+                def create_process_server(name)
```

Comment:
Why the complexity ? The test_remote_process.rb test file was able to start a server and connect to it with a lot less setup code (which you copied but are not using here)